mathics/builtin/testing_expressions/string_tests.py

```diff
@@ -0,0 +1,287 @@
+     = True
+
+    >> StringMatchQ["abc", "abd"]
+     = False
+
+    >> StringMatchQ["15a94xcZ6", (DigitCharacter | LetterCharacter)..]
+     = True
+
+    Use StringMatchQ as an operator
+    >> StringMatchQ[LetterCharacter]["a"]
+     = True
+    """
+
+    attributes = A_LISTABLE | A_PROTECTED
+
+    options = {
+        "IgnoreCase": "False",
+        "SpellingCorrections": "None",
+    }
+
+    rules = {
+        "StringMatchQ[patt_][expr_]": "StringMatchQ[expr, patt]",
+    }
+    summary_text = "test whether a string matches a pattern"
+
+    def eval(self, string, patt, evaluation: Evaluation, options: dict):
+        "StringMatchQ[string_, patt_, OptionsPattern[%(name)s]]"
+        py_string = string.get_string_value()
+        if py_string is None:
+            evaluation.message(
+                "StringMatchQ",
+                "strse",
+                Integer1,
+                Expression(SymbolStringMatchQ, string, patt),
+            )
+            return
+
+        re_patt = to_regex(
+            patt, show_message=evaluation.message, abbreviated_patterns=True
+        )
+        if re_patt is None:
+            evaluation.message(
+                "StringExpression",
+                "invld",
+                patt,
+                Expression(SymbolStringExpression, patt),
+            )
+            return
+
+        re_patt = anchor_pattern(re_patt)
+
+        flags = re.MULTILINE
+        if options["System`IgnoreCase"] is SymbolTrue:
+            flags = flags | re.IGNORECASE
+
+        if re.match(re_patt, py_string, flags=flags) is None:
+            return SymbolFalse
+        else:
+            return SymbolTrue
+
+
+class StringQ(Test):
+    """
+    <url>
+    :WMA link:
+    https://reference.wolfram.com/language/ref/StringQ.html</url>
+    <dl>
+      <dt>'StringQ[$expr$]'
+      <dd>returns 'True' if $expr$ is a 'String', or 'False' otherwise.
+    </dl>
+
+    >> StringQ["abc"]
+     = True
+    >> StringQ[1.5]
+     = False
+    >> Select[{"12", 1, 3, 5, "yz", x, y}, StringQ]
+     = {12, yz}
+    """
+
+    summary_text = "test whether an expression is a string"
+
+    def test(self, expr) -> bool:
+        return isinstance(expr, String)
+
+
+class SyntaxQ(Builtin):
+    """
+    <url>
+    :WMA link:
+    https://reference.wolfram.com/language/ref/SyntaxQ.html</url>
+    <dl>
+      <dt>'SyntaxQ["string"]'
+      <dd>returns 'True' if "string" corresponds to a syntactically correct input for a Mathics3 expression, or 'False' otherwise.
+    </dl>
+
+    >> SyntaxQ["a[b"]
+     = False
+
+    >> SyntaxQ["a[b]"]
+     = True
+    """
+
+    # FIXME:
+    # replace messages[string] with the below.
+    # Better is to have this kind of thing done a function for doing this.
+    # Same things for "noopt" message which does not exist yet.
+    messages = {"string": "String expected at position `1` in `2`."}
+
+    summary_text = (
+        "test whether a string is as yntactically-correct a Mathics3 expression"
```

Comment:
Typo: `syntactically`

Reply:
Thanks! Should be fixed now. 

BTW I don't know why this reminded me, but a long time ago I asked you in slack about packages that would be easy to get running. Your reply in slack history has been long removed. Do you remember what package or packages you mentioned?

Reply:
> BTW I don't know why this reminded me, but a long time ago I asked you in slack about packages that would be easy to get running. Your reply in slack history has been long removed. Do you remember what package or packages you mentioned?

I only documented this one in the Wiki:
- https://github.com/axkr/symja_android_library/wiki/context#wl-packages